### PR TITLE
Adding a new flow_permission: send_email_about_flow_page

### PR DESCRIPTION
**Pull request by dzhuang**
Preparing for #237

### CODE REVIEW

File: relate/urls.py [hunk 1]
```diff
@@ -391,6 +391,15 @@
         "/$",
         course.flow.finish_flow_session_view,
         name="relate-finish_flow_session_view"),
+    url(r"^course"
+        "/" + COURSE_ID_REGEX +
+        "/flow-session"
+        "/(?P<flow_session_id>[0-9]+)"
+        "/(?P<ordinal>[0-9]+)"
+        "/page_interaction_email"
```
Comment (inducer) [inline relate/urls.py]: Can you make that `page-interaction-email`? Just because all the other URL bits use dashes.

File: course/constants.py [hunk 2]
```diff
@@ -368,6 +368,8 @@ class flow_permission:  # noqa
     set_roll_over_expiration_mode = "set_roll_over_expiration_mode"
     see_session_time = "see_session_time"
     lock_down_as_exam_session = "lock_down_as_exam_session"
+    send_interact_email_to_course_staff =\
+        "send_interact_email_to_course_staff"
```
Comment (inducer) [inline course/constants.py]: Not sure I get why this needs to be per-page, rather than one of the (new) general `participation_permission`s?

Comment (dzhuang) [inline course/constants.py]: There should also be ones in the participation_permission(s), I will do that latter. The permission here is to restrict the usage of pages, I don't want to receive emails from students for all questions for all the time.

File: relate/urls.py (hunk 1 above)
Comment (dzhuang) [inline relate/urls.py]: ok

File: course/constants.py (hunk 2 above)
Comment (inducer) [inline course/constants.py]: Please also add that to the docs for `flow_permission`.

Comment (dzhuang) [inline course/constants.py]: Is that ok? I wish you can polish the naming of the code in order not to cause misunderstanding. Sorry for my bad English.

Comment (inducer) [inline course/constants.py]: How about `send_email_about_flow_page`?

Comment (dzhuang) [inline course/constants.py]: Thanks, updated.

File: course/templates/course/flow-page.html [hunk 3]
```diff
@@ -284,6 +284,55 @@
...
+          >
+              {% trans "Grant exception" %}
+          </a>
+
+      <a class="btn btn-primary btn-xs relate-btn-xs-vert-spaced"
+         href="{% url 'relate-grade_flow_page' course.identifier flow_session.id ordinal %}"
+         role="button"
+         target="_blank"
+      >
+        {% trans "View in grading page" %}
```
Comment (inducer) [inline course/templates/course/flow-page.html]: Sorry that these comments come out so piece-by-piece, but I only just discovered (ahem) this bit of the PR. :)

The permissions for this need to be a bit finer-grained.

* Reopen is `pperm.reopen_flow_session`.
* Grant exception is `pperm.grant_exception`.
* Seeing the grading page is `pperm.view_gradebook`.

You actually don't need to pass a flag from Python to the template. Just

```
{% if pperm.XYZ %}
{% endif %}
```
should do the trick.

Comment (dzhuang) [inline course/templates/course/flow-page.html]: Never say sorry about that. I really like the opportunity to improve the code. Thanks for the information.   Updated.

**Comment by inducer** [comment]
Looks good. Thanks!

**Pull request merged**